### PR TITLE
Support source security group id ingress access to cluster

### PR DESCRIPTION
**Pull request by sclausson**

### CODE REVIEW

**Comment by sclausson** [comment]
Plan runs ok locally, so not sure why/where the automated tests are failing.

File: modules/consul-cluster/main.tf [hunk 1]
```diff
@@ -95,6 +95,7 @@ resource "aws_security_group" "lc_security_group" {
 }
 
 resource "aws_security_group_rule" "allow_ssh_inbound" {
+  count       = "${length(var.allowed_inbound_cidr_blocks) >= 1 ? 1 : 0}"
```
Comment (brikis98) [inline modules/consul-cluster/main.tf]: Should this be `allowed_ssh_cidr_blocks`?

File: modules/consul-cluster/variables.tf [hunk 2]
```diff
@@ -24,6 +24,11 @@ variable "allowed_inbound_cidr_blocks" {
   type        = "list"
 }
 
+variable "allowed_inbound_security_group_ids" {
+  description = "A list of security group IDs that will be allowed to connect to Consul"
+  type        = "list"
```
Comment (brikis98) [inline modules/consul-cluster/variables.tf]: For backwards compatibility, should probably default this to an empty list.

File: modules/consul-security-group-rules/main.tf [hunk 3]
```diff
@@ -92,3 +101,101 @@ resource "aws_security_group_rule" "allow_dns_udp_inbound" {
   security_group_id = "${var.security_group_id}"
 }
 
+resource "aws_security_group_rule" "allow_server_rpc_inbound_from_security_group_ids" {
+  count                    = "${length(var.allowed_inbound_security_group_ids)}"
```
Comment (brikis98) [inline modules/consul-security-group-rules/main.tf]: Is `allowed_inbound_security_group_ids` a new input variable for the consul-security-group-rules module? If so, don't you need to add it to its `vars.tf` and pass it in from the `consul-cluster` module?

Comment (sclausson) [reply modules/consul-security-group-rules/main.tf]: Good catch. I have made these changes.

File: modules/consul-cluster/variables.tf (hunk 2 above)
Comment (sclausson) [reply modules/consul-cluster/variables.tf]: I've now set this to default = []

File: modules/consul-cluster/main.tf (hunk 1 above)
Comment (sclausson) [reply modules/consul-cluster/main.tf]: Good catch. I've corrected this.

**Comment by sclausson** [comment]
I think this should be ok now.

File: modules/consul-security-group-rules/variables.tf [hunk 4]
```diff
@@ -12,6 +12,11 @@ variable "allowed_inbound_cidr_blocks" {
   type        = "list"
 }
 
+variable "allowed_inbound_security_group_ids" {
+  description = "A list of security group IDs that will be allowed to connect to Consul"
+  type        = "list"
```
Comment (brikis98) [inline modules/consul-security-group-rules/variables.tf]: This should probably default to an empty list for backwards compatibility

**Comment by sclausson** [comment]
@brikis98 I've made one final requested change and added a default value to `allowed_inbound_security_group_ids` in `consul-security-group-rules/variables.tf` I think this is ready to merge.

**Comment by brikis98** [comment]
Fantastic, thank you!

**Comment by brikis98** [comment]
https://github.com/hashicorp/terraform-aws-consul/releases/tag/v0.0.4

**Pull request merged**